pkg/webhooks/kafkatopic_validator_test.go

```diff
@@ -72,6 +74,141 @@ func newMockClients(cluster *v1beta1.KafkaCluster) (runtimeClient.WithWatch, kaf
+					Name:              "test-topic",
+					Partitions:        2,
+					ReplicationFactor: 1,
+					Config:            map[string]string{"testConfKey": "testConfVal"},
+					ClusterRef:        v1alpha1.ClusterReference{},
+				},
+			},
+			expectedErrors: []string{TopicManagedByAnnotationKey},
+		},
+		{
+			testName: "topic replication factor is different and managedBy koperator",
+			kafkaTopic: v1alpha1.KafkaTopic{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{TopicManagedByAnnotationKey: TopicManagedByKoperatorAnnotationValue},
+				},
+				Spec: v1alpha1.KafkaTopicSpec{
+					Name:              "test-topic",
+					Partitions:        2,
+					ReplicationFactor: 5,
+					Config:            map[string]string{"testConfKey": "testConfVal"},
+					ClusterRef:        v1alpha1.ClusterReference{},
+				},
+			},
+			expectedErrors: []string{"replication"},
+		},
+		{
+			testName: "topic partition is different and managedBy koperator",
+			kafkaTopic: v1alpha1.KafkaTopic{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{TopicManagedByAnnotationKey: TopicManagedByKoperatorAnnotationValue},
+				},
+				Spec: v1alpha1.KafkaTopicSpec{
+					Name:              "test-topic",
+					Partitions:        5,
+					ReplicationFactor: 1,
+					Config:            map[string]string{"testConfKey": "testConfVal"},
+					ClusterRef:        v1alpha1.ClusterReference{},
+				},
+			},
+			expectedErrors: []string{"partition"},
+		},
+
+		{
+			testName: "topic partition and replication is different and not managedBy koperator",
+			kafkaTopic: v1alpha1.KafkaTopic{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{TopicManagedByAnnotationKey: TopicManagedByKoperatorAnnotationValue},
+				},
+				Spec: v1alpha1.KafkaTopicSpec{
+					Name:              "test-topic",
+					Partitions:        5,
+					ReplicationFactor: 5,
+					Config:            map[string]string{"testConfKey": "testConfVal"},
+					ClusterRef:        v1alpha1.ClusterReference{},
+				},
+			},
+			expectedErrors: []string{"replication", "partition", TopicManagedByAnnotationKey},
+		},
+		{
+			testName: "topic configuration is different and managedBy koperator",
+			kafkaTopic: v1alpha1.KafkaTopic{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{TopicManagedByAnnotationKey: TopicManagedByKoperatorAnnotationValue},
+				},
+				Spec: v1alpha1.KafkaTopicSpec{
+					Name:              "test-topic",
+					Partitions:        2,
+					ReplicationFactor: 1,
+					Config:            map[string]string{"testConfKey": "testConfVal1"},
+					ClusterRef:        v1alpha1.ClusterReference{},
+				},
+			},
+			expectedErrors: []string{"its configuration must be"},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.testName, func(t *testing.T) {
+			t.Parallel()
+			fieldErrorList, err := kafkaTopicValidator.checkKafka(context.Background(), &testCase.kafkaTopic, cluster)
```

Comment:
Does this section of code suffer from the "loop var + closures + goroutines" gotcha from Go ?

I tried adding a `t.Log(testCase)` beneath this line and the output always has `testName` as `topic configuration is different and managedBy koperator` (the last test case).
```
=== NAME  TestCheckKafkaTopicExist/topic_configuration_is_same_and_managedBy_koperator
    kafkatopic_validator_test.go:204: {topic configuration is different and managedBy koperator {{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[managedBy:koperator] [] [] []} {test-topic 2 1 map[testConfKey:testConfVal1] { }} { }} [its configuration must be]}
=== NAME  TestCheckKafkaTopicExist/topic_replication_factor_is_different_and_managedBy_koperator
    kafkatopic_validator_test.go:204: {topic configuration is different and managedBy koperator {{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[managedBy:koperator] [] [] []} {test-topic 2 1 map[testConfKey:testConfVal1] { }} { }} [its configuration must be]}
=== NAME  TestCheckKafkaTopicExist/topic_partition_and_replication_is_different_and_not_managedBy_koperator
    kafkatopic_validator_test.go:204: {topic configuration is different and managedBy koperator {{ } {      0 {{0 0 <nil>}} <nil> <nil> map[] map[managedBy:koperator] [] [] []} {test-topic 2 1 map[testConfKey:testConfVal1] { }} { }} [its configuration must be]}
```

Reply:
Looks to me like a proper full name.
`TestCheckKafkaTopicExist/topic_configuration_is_same_and_managedBy_koperator`
Not sure what you mean exactly.

Reply:
Oh, I see, I think after the for there should be a 
```go
testCase := testCase
```
line which fixes this.
(I use a linter that alerts on this.)

Thanks for catching Mihai.